12_glob_lab/globlab.txt

```diff
@@ -1,29 +1,33 @@
-Glob Lab.
+Glob La:b.
 https://www.tocode.co.il/bundles/unix1/lessons/globlab
 
 1. Type in a glob pattern to match:
 
   1. All files whose name starts with a lowercase letter
-  Answer: 
+  Answer: echo [a-z]*
 
   2. All files whose name starts with an uppercase letter
-  Answer:
+  Answer:[A-Z]*
 
   3. All files whose name ends in an "a", "e", "i", "o" or "u"
-  Answer:
+  Answer:echo [aeiou]
 
   4. All files whose name DOES NOT end in an "a", "e", "i", "o" or "u"
-  Answer:
+  Answer:echo [!aeiou]
 
   5. All files with extension ".c" that are in a folder named "src"
-  Answer:
+  Answer:echo *.c
 
   6. All files with extension ".c" under any folder
-  Answer:
+  Answer:find . -name "*.c" -type f
+
 
 2. Write a command that would delete all files ending with a ".png" under any subdirectory.
+sanswer : find . -name "*.png" -type f -delete
+
 
 3. The command "ls -l" can take the name of a file or directory and show details on them. From within "wildcards" folder run "ls -l [A-Z]*". What was printed? Why?
+answer: for each folder under wildcards we get all the data of the files there, related to the user permissions and time creation..
 
 4. Search in "man ls" for an option to show only directory data so the command "ls _____ D*" should print only the word Doc
-
+answer: find . -name "D*" -ls
```

Comment:
The command you typed shows all information for all files and directories starting with a D recursively, but the question asked about directory information only. What we wanted to see is which folders actually match `D*` without seeing the files inside them.

A simple way is to use `-d` switch to ls: `ls -d D*`. This shows Doc without showing the files inside